contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol

```diff
@@ -372,47 +389,119 @@ contract FillQuoteTransformer is
+        // If it is a Bridge order we fill this directly
+        // rather than filling via 0x Exchange
+        if (order.makerAssetData.readBytes4(0) == ERC20_BRIDGE_PROXY_ID) {
+            uint256 outputTokenAmount = LibMathV06.getPartialAmountFloor(
+                takerAssetFillAmount,
+                order.takerAssetAmount,
+                order.makerAssetAmount
+            );
+            (bool success, bytes memory data) = address(_implementation).delegatecall(
+                abi.encodeWithSelector(
+                    this.fillBridgeOrder.selector,
+                    order,
+                    takerAssetFillAmount,
+                    outputTokenAmount,
+                    state,
+                    makerToken
+                )
+            );
+            if (success) {
+                results = abi.decode(data, (FillOrderResults));
             }
-        } catch (bytes memory) {
             // Swallow failures, leaving all results as zero.
+        } else {
+            // Emit an event if we do not have sufficient ETH to cover the protocol fee.
+            if (state.ethRemaining < state.protocolFee) {
+                emit ProtocolFeeUnfunded(state.ethRemaining, state.protocolFee);
+                return results;
+            }
+            try
+                exchange.fillOrder
+                    {value: state.protocolFee}
+                    (order, takerAssetFillAmount, signature)
+                returns (IExchange.FillResults memory fillResults)
+            {
+                results.makerTokenBoughtAmount = fillResults.makerAssetFilledAmount;
+                results.takerTokenSoldAmount = fillResults.takerAssetFilledAmount;
+                results.protocolFeePaid = fillResults.protocolFeePaid;
+                // If the taker fee is payable in the taker asset, include the
+                // taker fee in the total amount sold.
+                if (isTakerFeeInTakerToken) {
+                    results.takerTokenSoldAmount =
+                        results.takerTokenSoldAmount.safeAdd(fillResults.takerFeePaid);
+                }
+            } catch (bytes memory) {
+                // Swallow failures, leaving all results as zero.
+            }
         }
     }
 
+    /// @dev Attempt to fill an ERC20 Bridge order. If the fill reverts,
+    ///      or the amount filled was not sufficient this reverts.
+    /// @param order The bridge order to fill.
+    /// @param inputTokenAmount How much taker asset to fill clamped to the available balance.
+    /// @param outputTokenAmount How much maker asset to receive.
+    /// @param makerToken The maker token.
+    function fillBridgeOrder(
+        IExchange.Order calldata order,
+        uint256 inputTokenAmount,
+        uint256 outputTokenAmount,
+        FillState calldata state,
+        IERC20TokenV06 makerToken
+    )
+        external
+        returns (FillOrderResults memory results)
```

Comment:
Do we even really need a return value here? Can't we figure this all out in `fillOrder()` and avoid encoding/decoding a return value? Would also prefer to keep this function simple to audit since it executes as a delegetecall.